### ISSUE TITLE
Lack of Lego Pump control in version 1.2.3

### ISSUE THREAD

**Issue by MattH688** [HenriquesLab/NanoJ-Fluidics#1]
Hello,

As of version 1.2.3 a commit has been submitted with a complied ".jar" which fails to communicate with the Lego pumps but appears to send commands to the Arduino. I am currently unsure if the issue results from version 1.2.2 as there was no complied ".jar" on that release. So far I have been using the last known working release version 1.2.1.

Many thanks,
MattH688

**Comment by palmada** [comment]
Hi @MattH688,

Did you just use the JAR? Or the packaged binaries in https://github.com/HenriquesLab/NanoJ-Fluidics/tree/master/PackagedBinaries?

Kind regards,
Pedro

**Comment by MattH688** [comment]
Hi @palmada,

I downloaded the entire project "Manual, source-code and binaries for the NanoJ-Fluidics project". I attempted the ".jar" and have attempted a recompiled v1.2.3 this morning. I would like to update my original issue as I have realised direct Lego pump control works* but sequence does not appear to send commands to the correct pump. I have checked and confirmed the messages to the Arduino are correct. I believe it is an issue somewhere with the sequence java files assigning the correct pump index. 

Many thanks,
Matthew

*As expected except for the hard coded max speed for Lego pumps as well as failing to stop after the sequence in v1.2.1 which I have fixed in my fork. Happy to pull into yours but ideally need sequences fixed first on your master.
